tests/unit/customizations/s3/test_cp_command.py

```diff
@@ -60,8 +60,45 @@ def test_trailing_slash_appended(self):
         # The only operation we should have called is PutObject.
         self.assertEqual(len(self.operations_called), 1, self.operations_called)
         self.assertEqual(self.operations_called[0][0].name, 'PutObject')
-        self.assertEqual(self.operations_called[0][1]['key'], 'foo.txt')
-        self.assertEqual(self.operations_called[0][1]['bucket'], 'bucket')
+        self.assertEqual(self.operations_called[0][1]['Key'], 'foo.txt')
+        self.assertEqual(self.operations_called[0][1]['Bucket'], 'bucket')
+
+    def test_upload_grants(self):
+        full_path = self.files.create_file('foo.txt', 'mycontent')
+        cmdline = ('%s %s s3://bucket/key.txt --grants read=id=foo '
+                   'full=id=bar readacl=id=biz writeacl=id=baz' %
+                   (self.prefix, full_path))
+        self.parsed_responses = \
+            [{'ETag': '"c8afdb36c52cf4727836669019e69222"'}]
+        self.run_cmd(cmdline, expected_rc=0)
+        # The only operation we should have called is PutObject.
+        self.assertEqual(len(self.operations_called), 1,
+                         self.operations_called)
+        self.assertEqual(self.operations_called[0][0].name, 'PutObject')
+        self.assertEqual(self.operations_called[0][1]['Key'], 'key.txt')
```

Comment:
This might read a little better if you just grab the dict and use `assertDictEqual`, e.g.:

```
kwargs = self.operations_called[0][1]
self.assertDictEqual(kwargs, {'Key': 'key.txt', 'Bucket': '...', ...})
```

It would also catch the case where additional params showed up that weren't suppose to be there.